### PR TITLE
Throttle Fix

### PR DESCRIPTION
**Pull request by nerdgilbert**
## Major changes
**Fixes: https://github.com/OpenROV/openrov-software/issues/630**
- [x] Updates variables to use function returns
- [x] Fixes listen() bug in rovpilot that did not fire correctly

@BrianAdams if you can look over this, that would be great. I don't know how idiomatic it is, but it works now and the previous way didn't. Is there a better way to do this?

### CODE REVIEW

File: src/plugins/rovpilot/public/js/rovpilot.js [hunk 1]
```diff
@@ -379,11 +379,7 @@
       //to deal with the clocks not being in sync between the computer and the ROV.
       var self = this;
 
-      //Initial sync of state
-      self.cockpit.rov.emit('plugin.rovpilot.getState', function(state) {
```
Comment (nerdgilbert) [inline src/plugins/rovpilot/public/js/rovpilot.js]: In MOCK mode, this fired but the callback function never fired. It fires when you "wait" a little bit and then emit this.

@BrianAdams thoughts?

Comment (BrianAdams) [inline src/plugins/rovpilot/public/js/rovpilot.js]: So, the powerLevel is not a Global setting with the ROV but instead just a client setting. So rovpilot.js is actually the owner of the state (thus why it sets the default value).  No need to try and sync state with the ROV

File: src/plugins/new-ui/public/webcomponents/throttle.html [hunk 2]
```diff
@@ -280,22 +280,24 @@
...
         maxThrottle: {
           type: Number,
-          value: 100
+          value: function(){return 100;}
         },
-        power: {
+        powerLevel: {
           type: Number,
-          value: 0
+          value: function(){return 0;}
```
Comment (BrianAdams) [inline src/plugins/new-ui/public/webcomponents/throttle.html]: I imagine the private vs shared variables may fix the issue all by itself.

File: src/plugins/rovpilot/public/js/rovpilot.js (hunk 1 above)
Comment (BrianAdams) [inline src/plugins/rovpilot/public/js/rovpilot.js]: Not actually sure why this code was here. Either it was left over from a prior refactor or I missed the a change where we moved the state of power level to the ROV which would not make sense.

**Comment by nerdgilbert** [comment]
Okay, I will make changes as per your recommendation.

**Comment by nerdgilbert** [comment]
This should work. The main problem was in the way the svg elements were getting drawn at startup. The elements weren't ready when they were being set so they just defaulted to the SVG declarations.

**Pull request merged**